msgraph/core/middleware/middleware.py

```diff
@@ -31,9 +32,14 @@ def add_middleware(self, middleware):
 
     def send(self, request, **kwargs):
 
-        if not hasattr(request, 'context'):
-            headers = request.headers
-            request.context = RequestContext(dict(), headers)
+        middleware_control_json = request.headers.pop('middleware_control', None)
+        if middleware_control_json:
+            middleware_control = json.loads(middleware_control_json)
+        else:
+            middleware_control = dict()
+
+        headers = request.headers
+        request.context = RequestContext(middleware_control, headers)
```

Comment:
Nit: 
You can pass `request.headers` to RequestContext then line 41 will be unnecessary.